funfact/algorithm.py

```diff
@@ -12,7 +13,7 @@
 def factorize(
     tsrex, target, lr=0.1, tol=1e-6, max_steps=10000, optimizer='Adam',
     loss='mse_loss', nvec=1, stop_by='first', returns='best',
-    checkpoint_freq=50, **kwargs
+    checkpoint_freq=50, dtype='target', **kwargs
```

Comment:
How about `dtype=None`, and then infer `dtype` if it is None? This should be as intuitive as `'target'`, and since infer dtype from target is the only special case here, we don't really need to use a string to name it.

Reply:
Done